transcom/mymove
# [MB-16271] TOO can approve or reject address change and add remarks

## [MB-16271](https://dp3.atlassian.net/browse/MB-16271)

## Summary

> Is there anything you would like reviewers to give additional scrutiny?

Yes. There was a previous bug fixed in this PR where the "Review request" link was not clickable on browsers other than chrome. This is because the alert containing the button link was inside of a disabled fieldset. To circumvent this, the markup has been changed. "Delivery location" is no longer the fieldset's legend, instead it is text placed before the fieldset so that the alert could be placed between the text and the fieldset.

~~Additionally, though many tests have been added, this PR is currently dropping test coverage by less than 0.1%. Since the bulk of this new code is react query mutation logic, and this project doesn't test those, the overall coverage is dropping. I think I understand why folks have steered clear in the past. Mocking a mutation's onSubmit and onFailure in tests is pretty gnarly. If time permits, I'll try to demonstrate on this PR. If it works, I'll remove this note.~~ We did it 🎉 

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.

### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/run-storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/about/application-setup/milmove-local-client/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/about/development/testing)

### How to test

1. Start up the app
2. Login as a TOO
3. Open the CRQST1 move loaded as seed data
4. Click edit shipment
5. Click review request in the alert under delivery location
6. fill out the form in the modal
7. note that the database entry for that ShipmentAddressUpdate has been updated
8. note that the modal closes
9. note that the alerts previously on the shipment form are now gone
10. note that a success alert now appears
11. note that the success alert is close-able

### Frontend


### Backend

- No changes

## Screenshots


https://github.com/transcom/mymove/assets/15805554/42ec7e7b-b0a6-4e56-b0ca-8fa58e6fe306

<img width="299" alt="image" src="https://github.com/transcom/mymove/assets/15805554/b4d5a99b-323d-4b3b-985c-a56e29bcdc2b">


[MB-16271]: https://dp3.atlassian.net/browse/MB-16271?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
